src/shims/unix/socket.rs

```diff
@@ -22,15 +29,71 @@ impl FileDescription for SocketPair {
     ) -> InterpResult<'tcx, io::Result<()>> {
         Ok(Ok(()))
     }
+
+    fn read<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        bytes: &mut [u8],
+        _tcx: TyCtxt<'tcx>,
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        let request_byte_size = bytes.len();
+        let data_size = self.readbuf.borrow_mut().len();
+
+        // FIXME: blocking is currently not supported
+        if data_size == 0 {
+            throw_unsup_format!("socketpair read: blocking isn't supported yet");
+        }
+        if request_byte_size == 0 {
+            return Ok(Ok(0));
+        }
```

Comment:
do this before the available bytes check, because reading zero bytes is always fine.